### ISSUE TITLE
no-document-write: Rule is triggered by any function named "write"

### ISSUE THREAD

**Issue by nfriend** [Microsoft/tslint-microsoft-contrib#108]
If I create an object with a `write` method, and the `no-document-write` rule is `true`, calling my `write` method results in a `Forbidden call to document.write` linting error:

```typescript
declare var myObject;

// test.ts[1, 1]: Forbidden call to document.write
myObject.write();
```

Expected result: the `no-document-write` rule should only trigger a linting error when the `write` method call is made to the `document` object.

**Comment by nfriend** [comment]
Now that I look at the source for the `no-document-write` rule, I'm noticing that it specifically targets the `document` object **or** variables with type `any`:

```typescript
if (leftSideType.flags === ts.TypeFlags.Any || typeAsString === 'Document') {
    if (functionName === 'write') {
        this.addFailure(this.createFailure(leftSide.getStart(), leftSide.getWidth(), Rule.WRITE_FAILURE));
    }
    else {
        this.addFailure(this.createFailure(leftSide.getStart(), leftSide.getWidth(), Rule.WRITELN_FAILURE));
    }
}
```

In this case, perhaps the error message should mention variables of type `any` will trigger this condition?

**Comment by nfriend** [comment]
After playing around with this rule for a while, I've managed to produce a better example that illustrates the issue I'm having. See #109.  However, I think my comment about this rule's error message is still relevant, so I'll leave this issue open for now.

**Comment by HamletDRC** [comment]
Hi @nfriend . You are seeing two different issues inherent in the tslint project. 
1. tslint scans files one file at a time and does not compile the entire project. This means that any types defined in other files or other d.ts files will not be seen and will be reported as "any". So by design, tslint is going to give you "any" for almost all your references, even when the tsc compiler would give you a real type. 
2. tslint's typechecker does not work correctly and reports "any" for many references even within a single file. There are times when all the types are defined within a single source file and tslint still reports "any".

So these 2 reasons are why we chose to implement the rule to be very strict and consider "any" to be the same as "Document". Honestly, I think that making the rule more flexible would also make the rule mostly useless because the type is so rarely "Document".

Is this explanation sufficient to close the bug as "as designed" ?

**Comment by nfriend** [comment]
Thanks for the detailed explanation, @HamletDRC.  Without having the ability to definitively know types, it makes sense that the rule was implemented so strictly.
